monai/networks/layers/factories.py

```diff
@@ -8,8 +8,169 @@
+        """
+        
+        if not isinstance(factory_name, str):
+            raise ValueError("Factories must be selected by name")
+            
+        fact = self.factories[factory_name.upper()]
+        return fact(*args)
+
+    def __getitem__(self, args):
+        """
+        Get the given name or name/arguments pair. If `args` is a callable it is assumed to be the constructor
+        itself and is returned, otherwise it should be the factory name or a pair containing the name and arguments.
+        """
+        
+        # `args` is actually a type or constructor
+        if callable(args):
+            return args
+        
+        # `args` is a factory name or a name with arguments
+        if isinstance(args, str):
+            name_obj, args = args, ()
+        else:
+            name_obj, *args = args
+
+        return self.get_constructor(name_obj, *args)
+
+    def __getattr__(self, key):
+        """
+        If `key` is a factory name, return it, otherwise behave as inherited. This allows referring to factory names
+        as if they were constants, eg. `Fact.FOO` for a factory Fact with factory function foo.
+        """
+        
+        if key in self.factories:
+            return key
+
+        return super().__getattr__(key)
+
+    
+def split_args(args):
+    """
+    Split arguments in a way to be suitable for using with the factory types. If `args` is a name it's interpreted 
+    """
+    
+    if isinstance(args,str):
+        return args,{}
+    else:
+        name_obj,args=args
+        
+        if not isinstance(name_obj,(str,Callable)) or not isinstance(args,dict):
+            raise ValueError("Layer specifiers must be single strings or pairs of the form (name/object-types, argument dict)")
+            
+        return name_obj, args
+    
+
+### Define factories for these layer types
+    
+Dropout = LayerFactory()
+Norm = LayerFactory()
+Act = LayerFactory()
+Conv = LayerFactory()
+
+
+@Dropout.factory_function("dropout")
+def get_dropout_type(dim):
+    types = [nn.Dropout, nn.Dropout2d, nn.Dropout3d]
+    return types[dim - 1]
+
+
+@Norm.factory_function("instance")
+def get_instance_type(dim):
+    types = [nn.InstanceNorm1d, nn.InstanceNorm2d, nn.InstanceNorm3d]
+    return types[dim - 1]
+
+
+@Norm.factory_function("batch")
+def get_batch_type(dim):
+    types = [nn.BatchNorm1d, nn.BatchNorm2d, nn.BatchNorm3d]
+    return types[dim - 1]
+
+
+Act.add_factory_callable("relu", lambda: nn.modules.ReLU)
+Act.add_factory_callable("leakyrelu", lambda: nn.modules.LeakyReLU)
+Act.add_factory_callable("prelu", lambda: nn.modules.PReLU)
+
+
+@Conv.factory_function("conv")
+def get_conv_type(dim):
+    types = [nn.Conv1d, nn.Conv2d, nn.Conv3d]
+    return types[dim - 1]
+
+
+@Conv.factory_function("convtrans")
+def get_convtrans_type(dim):
+    types = [nn.ConvTranspose1d, nn.ConvTranspose2d, nn.ConvTranspose3d]
+    return types[dim - 1]
+
+
+### old factory functions
```

Comment:
As we are still at early stage of network development, I think we need to adapt all existing networks to new APIs and remove old APIs ASAP.
Thanks.